src/getCypressElementCoordinates.ts

```diff
@@ -46,6 +46,88 @@ function getPositionedCoordinates(
       return [x0 + width / 2, y0 + height / 2];
   }
 }
+/**
+ * Scrolls the given htmlElement into view on the page.
+ * The position the element is scrolled to can be customized with scrollBehavior.
+ */
+function scrollIntoView(
+  htmlElement: HTMLElement,
+  scrollBehavior: ScrollBehaviorOptions = "center"
+) {
+  let block: ScrollLogicalPosition;
+
+  if (scrollBehavior === "top") {
+    block = "start";
+  } else if (scrollBehavior === "bottom") {
+    block = "end";
+  } else {
+    block = scrollBehavior;
+  }
+
+  htmlElement.scrollIntoView({ block });
+}
+
+function getIframesPositionShift(element: HTMLElement) {
+  let currentWindow: Window | null = element.ownerDocument.defaultView;
+  const noPositionShift = {
+    frameScale: 1,
+    frameX: 0,
+    frameY: 0,
+  };
+
+  if (!currentWindow) {
+    return noPositionShift;
+  }
+
+  // eslint-disable-next-line prefer-const
+  const iframes = []
+
+  while (
+    currentWindow &&
+    currentWindow !== window.top
+  ) {
+    iframes.push(
+      // for cross origin domains .frameElement returns null so query using parentWindow
+      // but when running using --disable-web-security it will return the frame element
+      (currentWindow.frameElement as HTMLElement) ??
+        currentWindow.parent.document.querySelector("iframe")
```

Comment:
out of curiosity - is there any situation in which this fallback would actually kick in? note that if it does it might not return the correct iframe as there might be multiple iframes on the page and this doesn't account for that

Reply:
Yep when running with disable-web-security it follows the original .frameElement. 

I didn’t get the last part. This prop returns the frame where window is embedded. It means that it returns the upper frame which is the only that contains this window. 

If think this way it may be really possible to remove this branching, but what exact problem do you see here?

Reply:
> Yep when running with disable-web-security it follows the original .frameElement.

I was the one who has checked how this behaves with disable-web-security and to my slight surprise, it just has returned the "correct" `.frameElement` despite `.frameElement` having to return `null` for cross-origin iframes. So it is as web-security is also responsible for guarding access to `.frameElement` - so it returns `null` normally and if we disable it starts to return the correct element.

So using disable-web-security (according to my tests) doesn't influence the logic anyhow - and from what I understand we shouldn't be even able to select cross-origin element in Cypress (when not using disable-web-security) as it's running in the real browser's context so it would just not allow us to do that. 

> I didn’t get the last part. This prop returns the frame where window is embedded. It means that it returns the upper frame which is the only that contains this window.

I was referring to the fallback logic (`currentWindow.parent.document.querySelector("iframe")`) - it can just queries the first iframe but the `currentWindow.parent.document` might contain more than one and thus we can select incorrect one using this.

> If think this way it may be really possible to remove this branching, but what exact problem do you see here?

I just usually try to avoid dead branches of the code and was really curious if this is dead or not (the `??` part). I might very well just miss something here. And, as mentioned, even if it's not dead I believe that it might lead to incorrect results at times.

Reply:
Makes sense. So it needs to be processed by looking for iframe which content window is equal to current frame, yeah it looks correct. 

Feel free to open PR, I am on vacation right now so won’t code — but I can review PR and merge if everything is fine

Reply:
The question is - how to write a test for this? Under what circumstances are we able to "fail" reading the `currentWindow.frameElement`?

Reply:
In any case except cypress is running over chrome with disabled web security I think

Reply:
In a regular scenario (without disabled web security) - we can't even get access to cross-origin frames. I've just re-tested this using this repo:
> Uncaught DOMException: Blocked a frame with origin "https://example.cypress.io" from accessing a cross-origin frame.

So I **think** (but not entirely sure) that it's not possible to get access to an element from some inner cross-origin so this entire logic just can't be executed on such an element (no cypress command can).

However, I see that a regular relation between embedding context and an embedded iframe is somehow hacked for the AUT in Cypress. How do you even achieve that? 🤔 I have never actually thought about it as our tests are running against localhost.

And note - a cross-origin AUT has `.frameElement === null`. So this logic is actually not dead 👍 As it handles the last iframe on the stack (the AUT) and it just happens that the AUT is the first one (at least for now 😉 ). 

If I have made any mistake in my whole reasoning - please let me know.

Reply:
Cypress proxies response and change the origin of content so the origin will also the same. The whole cypress is build over making iframe same origin.

Reply:
Cool, that's good to know! Now when you have said that I think I've seen your injection code when debugging - just didn't connect the dots.

Either way - would you agree then that this fallback logic is currently only executed when looking for the `frameElement` of the AUT? Or are there any other cases as well that come to your mind?

Reply:
Yes it looks like it exists for aut or the edge case, but you are right it’s mainly for the cypress aut